HousingManagementSystemApi.Tests/UseCasesTests/RetrieveAddressesUseCaseTests.cs

```diff
@@ -7,28 +7,40 @@
 using HousingManagementSystemApi.UseCases;
```

Comment:
Please add tests specifically for the filtering based only on tenure, i.e. addresses with valid asset type(s) and eligible and ineligible tenure types (at least a single test for each of these)